### PR TITLE
[6.1.0] Add and update STS content

### PR DESCRIPTION
**Pull request by divyaamunugama**
Related issue: https://github.com/wso2/product-is/issues/14908

### CODE REVIEW

File: en/docs/guides/identity-federation/configure-ws-trust.md [hunk 1]
```diff
@@ -46,141 +41,107 @@ To secure the Security Token Service:
...
-    
-	![add-new-trusted-service]({{base_path}}/assets/img/tutorials/add-new-trusted-service.png)
+## Configure the service provider
+
+The steps in this section should be followed if you are using Holder of Key as the [subject confirmation method]({{base_path}}/references/concepts/authentication/intro-ws-trust/#subject-confirmation).
+
+For more information, see [how to configure STS for obtain tokens with Holder-Of-Key subject confirmation]({{base_path}}/learn/configuring-sts-for-obtaining-tokens-with-holder-of-key-subject-confirmation).
+
+!!! Tip "Before you begin"
+    You must first [register a service provider]({{base_path}}/learn/adding-and-configuring-a-service-provider/#adding-a-service-provider).
```
Comment (divyaamunugama) [inline en/docs/guides/identity-federation/configure-ws-trust.md:104]:
```suggestion
    You must first [register a service provider]({{base_path}}//guides/applications/register-sp/).
```

File: en/docs/guides/identity-federation/configure-ws-trust.md [hunk 2]
```diff
@@ -46,141 +41,107 @@ To secure the Security Token Service:
...
-    
-	![add-new-trusted-service]({{base_path}}/assets/img/tutorials/add-new-trusted-service.png)
+## Configure the service provider
+
+The steps in this section should be followed if you are using Holder of Key as the [subject confirmation method]({{base_path}}/references/concepts/authentication/intro-ws-trust/#subject-confirmation).
+
+For more information, see [how to configure STS for obtain tokens with Holder-Of-Key subject confirmation]({{base_path}}/learn/configuring-sts-for-obtaining-tokens-with-holder-of-key-subject-confirmation).
+
+!!! Tip "Before you begin"
+    You must first [register a service provider]({{base_path}}/guides/applications/register-sp/).
```
Comment (divyaamunugama) [inline en/docs/guides/identity-federation/configure-ws-trust.md:104]:
```suggestion
<!-- For more information, see [how to configure STS for obtain tokens with Holder-Of-Key subject confirmation]({{base_path}}/learn/configuring-sts-for-obtaining-tokens-with-holder-of-key-subject-confirmation). -->

!!! Tip "Before you begin"
    You must first [register a service provider]({{base_path}}/guides/applications/register-sp/).
```

File: en/docs/guides/identity-federation/configure-ws-trust.md [hunk 3]
```diff
@@ -46,141 +41,105 @@ To secure the Security Token Service:
...
 
-        !!! note
-                The **Select Domain** drop-down lists many domains. The listed
-                **User Groups** can vary depending on the domain selected.
-        
+    !!! info
+        The next steps may vary per the security scenario you have chosen under point (5) above. Given below are the steps for the **UsernameToken** scenario.
 
-    	![select-domain]({{base_path}}/assets/img/tutorials/select-domain.png)
+    1. Select `ALL-USER-STORE-DOMAINS` as the domain.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-federation/configure-ws-trust.md:80]:
```suggestion
    1. Select **ALL-USER-STORE-DOMAINS** as the domain.
```

File: en/docs/guides/identity-federation/configure-ws-trust.md [hunk 4]
```diff
@@ -46,141 +41,105 @@ To secure the Security Token Service:
...
+    2. Select the role you created to grant permission to access a secured service. In this example, the `admin` role is used.
+        ![select-domain]({{base_path}}/assets/img/guides/ws-trust-select-domain.png)
 
-7.  Click **Finish**.
-8.  Click **Ok** on the confirmation dialog window that appears.
-9.  Click **Update** to complete the process.
+        !!! note
+            The **Select Domain** drop-down lists the available domains. The listed **User Groups** can vary depending on the domain selected from the drop-down.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-federation/configure-ws-trust.md:85]:
```suggestion
            The **Select Domain** list provides the available domains. The listed **User Groups** can vary depending on the domain selected from the list.
```

File: en/docs/guides/identity-federation/configure-ws-trust.md [hunk 5]
```diff
@@ -46,141 +41,105 @@ To secure the Security Token Service:
...
+            <tr class="odd">
+                <td><strong>Endpoint Address</strong></td>
+                <td>
+                    <div class="content-wrapper">
+                    <div>
+                        Enter the trusted relying party's <strong>endpoint address</strong>, which is the endpoint address of the Security Token Service.
+                        <p>The endpoint must be used as the service URL to which the token gets delivered by the STS client. Then select the public certificate imported. Tokens issued are encrypted using the public certificate of the trusted relying party. Therefore, the consumer who obtains this token to invoke the RP service will not be able to see the token.</p>
+                        <div class="admonition note">
+                            <p class="admonition-title">Note</p>
+                            Make sure to upload the certificate of the relying party to the trust store. For instructions, see <a href="https://is.docs.wso2.com/en/latest/deploy/security/create-new-keystores/#add-ca-signed-certificates-to-keystores">adding CA-signed certificates to keystores</a>.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-federation/configure-ws-trust.md:125]: Why did we use the direct link here? was the a reason?
@divyaamunugama

File: en/docs/guides/identity-federation/obtain-tokens-using-sts.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Configuring STS for Obtaining Tokens with Holder-Of-Key Subject Confirmation
+
+In this scenario, STS generates a Holder-Of-Key subject confirmation (symmetric key) and encrypts it with the public key of the relying party. This is included in the subject confirmation section of the SAML token, which is validated by the relying party. For this to work, the relying party's endpoint address and the corresponding public certificates needs to be registered.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-federation/obtain-tokens-using-sts.md:3]:
```suggestion
In this scenario, STS generates a Holder-Of-Key subject confirmation (symmetric key) and encrypts it with the public key of the relying party. This is included in the subject confirmation section of the SAML token, which is validated by the relying party. For this to work, the relying party's endpoint address and the corresponding public certificates need to be registered.
```

File: en/docs/guides/identity-federation/obtain-tokens-using-sts.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    The tokens issued are encrypted using the public key of the trusted relying party. Accordingly, even the client that obtains the token to send to the RP has no visibility to the included token.  
+    Example: Enter the endpoint address of the service that you are running.  
+
+    ![add-new-trusted-service]({{base_path}}/assets/img/guides/ws-trust-sts-config.png)
+
+6. Click **Apply**.
+
+A new trusted service is added to the service provider.
+
+You can delete any trusted service by clicking on the associated **Delete** link.
```
Comment (nilminiwso2) [inline en/docs/guides/identity-federation/obtain-tokens-using-sts.md:22]:
```suggestion
You can delete any trusted service by clicking the associated **Delete** link.
```

File: en/docs/includes/running-an-sts-client.md [hunk 8]
```diff
@@ -1,3 +1,31 @@
-## Running an STS client
+## Run a sample STS Client
 
+The following sample demonstrates the steps to run a Security Token Service (STS) client. The STS issues a security token to authenticate a user for your client application using WS-Trust protocols.
 
+!!! note "Before you begin"
+    You need to [configure WS-Trust]({{base_path}}/guides/identity-federation/configure-ws-trust/) for WSO2 Identity Server.
+
+To run the STS client
```
Comment (nilminiwso2) [inline en/docs/includes/running-an-sts-client.md:8]:
```suggestion
To run the STS client:
```

File: en/docs/includes/running-an-sts-client.md [hunk 9]
```diff
@@ -1,3 +1,31 @@
...
+
+1. Clone the [Identity Server samples](https://github.com/wso2/samples-is) repository.
+
+2. In the cloned directory, navigate to `SAMPLES-IS/sts/sts-client` directory to build the STS client.
```
Comment (nilminiwso2) [inline en/docs/includes/running-an-sts-client.md:12]:
```suggestion
2. In the cloned directory, navigate to the `SAMPLES-IS/sts/sts-client` directory to build the STS client.  
```

**Review by nilminiwso2 — APPROVED** [review]

**Review by nilminiwso2 — APPROVED** [review]

**Pull request merged**